commerce_coordinator/apps/titan/pipeline.py

```diff
@@ -60,14 +68,39 @@ def run_filter(self, edx_lms_user_id, payment_number=None, **kwargs):  # pylint:
         try:
             payment = api_client.get_payment(
                 edx_lms_user_id=edx_lms_user_id,
-                payment_number=payment_number
             )
         except HTTPError as exc:
             logger.exception("[GetTitanPayment] Payment %s not found for user: %s", payment_number, edx_lms_user_id)
             raise PaymentNotFound from exc
         payment_serializer = PaymentSerializer(data=payment)
         payment_serializer.is_valid(raise_exception=True)
-        return payment_serializer.data
+        payment = payment_serializer.data
+
+        order_uuid = kwargs.get('order_uuid')
+        if order_uuid and payment['order_uuid'] != kwargs['order_uuid']:
```

Comment:
Would it make sense to explicitly add `kwargs['order_uuid']` as an optional argument to this pipeline step, perhaps called `expected_order_uuid`?

Reply:
added `order_uuid` as optional param for this step. we are getting these values from a common serializer so renaming it to `expected_order_uuid ` will make it more messy. Although I have added this information in the doc string these parameters are expected values.

Reply:
Thank you!